modules/sc-mesh-secure-deployment/src/nats/mdm_agent.py

```diff
@@ -247,8 +260,13 @@ def start_service_discovery(self):
         """
         if not self.service_monitor.running:
             self.logger.debug("Start service monitor thread")
-
-            self.__service_monitor_thread.start()
+            if not self.__service_monitor_thread.is_alive():
+                self.__service_monitor_thread = threading.Thread(
+                    target=self.service_monitor.run
+                )
```

Comment:
I'd suggest using a ThreadPool or using a loop with a Queue instead of creating a new thread every time as it seems their memory is not released by Python when they stop :cactus:

Reply:
As tested by @saauvine, it seems that this issue no longer affects modern Python3 :snake: